File: app/src/main/java/com/nanicky/mediaviewer/VideoDetails.java
package com.nanicky.mediaviewer;

import android.graphics.Bitmap;

import java.util.Date;

public class VideoDetails {

    public Bitmap videoThumbNail;
    public String name;
    public String videoPath;
    public Date dateTaken;
    public Date dateModified;
    public Date dateAdded;
    public String resolution;
    public String mimeType;

    public VideoDetails(Bitmap videoThumbNail, String name, String videoPath, Date dateTaken, Date dateModified, Date dateAdded, String resolution, String mimeType) {
        this.videoThumbNail = videoThumbNail;
        this.name = name;
        this.videoPath = videoPath;
        this.dateTaken = dateTaken;
        this.dateModified = dateModified;
        this.dateAdded = dateAdded;
        this.resolution = resolution;
        this.mimeType = mimeType;
    }

    public VideoDetails(String name, String videoPath, Date dateTaken, Date dateModified, Date dateAdded, String resolution, String mimeType) {
        this.name = name;
        this.videoPath = videoPath;
        this.dateTaken = dateTaken;
        this.dateModified = dateModified;
        this.dateAdded = dateAdded;
        this.resolution = resolution;
        this.mimeType = mimeType;
    }
}

File: app/src/main/java/com/nanicky/mediaviewer/db/VideoDetails.kt
package com.nanicky.mediaviewer.db

import android.graphics.Bitmap
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import java.util.*

@Entity(tableName = "VideoDetails")
@TypeConverters(DateConverter::class)
data class VideoDetails(
        @PrimaryKey()
        var path: String,
        @Ignore
        var thumb: Bitmap?,
        var name: String,
        var dateTaken: Date,
        var dateModified: Date,
        var dateAdded: Date,
        var resolution: String,
        var duration: String,
        var mimeType: String
) {
    constructor() : this("", null, "",
            Date(), Date(), Date(),
            "", "", "")

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as VideoDetails
        return this.path != other.path
    }
}
